docs/adr/009_lns_affinity.md

```diff
@@ -0,0 +1,286 @@
+# 009. LNS sticky affinity over multiple sessions
+
+**Feature**: [#1475](https://github.com/Azure/iotedge-lorawan-starterkit/issues/1475)  
+
+**Authors**:
+
+**Status**: Proposed
+__________
+
+## Problem statement
+
+Consider the topology:
+
+```mermaid
+flowchart LR;
+  Device-->LBS1-->LNS1--1-->IoTHub;
+  Device-->LBS2-->LNS2--2-->IoTHub;
+```
+
+where LNS 1 and 2 make use of their IoT Edge Hub modules to connect to IoT Hub.
+
+IoT Hub limits active connections that an IoT device can have to one. Assuming that connection 1 is
+already open and a message from LNS2 arrives, IoT Hub will close connection 1 and open connection 2.
+Edge Hub on LNS1, will detect this and assume it's a transient network issue, therefore will try
+proactively to reconnect to IoT Hub. IoT Hub will now drop the connection 2 to re-establish the
+original connection 1.  
+
+This connection "ping-pong" will continue happening, negatively impacting the scalability due to the
+high costs of setting up/disposing the connections. From our load tests we observed that in this
+scenario we were not even able to connect more than 120 devices to two LNSs, while in a single LNS
+topology we could scale up to 900 devices without issues.
+
+## Out of scope
+
+- Deduplication strategies Mark and None: these strategies rely on multiple LNSs sending message.
+Potentially we could consider other workarounds for the IoT Hub limitation of a single connection
+per device but we find it acceptable for the Mark and None strategies to not be as scalable as the
+Drop strategy and will only document this limitation for potential users to be aware of.
+
+- LNS performs operations on behalf of a device/sensor and a concentrator/station. However since a
+concentrator can be connected to at most one LNS, there is no ping-pong happening with operations on
+stations.
+
+- Class C downstream messages using Direct Method:
+  - When we send a downstream message via the Function, the Function is responsible for choosing the
+    prererred gateway based on its existing knowledge of the preferred gateway. Due to that we can
+    be sure that we have the active connection and can therefore send downstream messages.
+  - Messages sent from the portal could result in a connection switch if they target a LNS other
```

Comment:
correction: we can actually limit the impact. We know if we are the connection owner for that device and can act accordingly. This is a very simple case and we should implement it (close the connection, if we don't own it).

Reply:
ok, I will add it back to in-scope then and added a task for it in #1473

Reply:
Resolving now.